tests/integration_tests/tests/agentless_tests/test_executions.py

```diff
@@ -603,6 +605,28 @@ def test_cancel_execution_and_then_force_cancel(self):
 
         self._assert_execution_cancelled(execution, deployment_id)
 
+    def test_execute_and_kill_execution(self):
+        """
+        Tests the kill execution option by asserting the execution pid doesn't
+        exist.
+        """
+        dsl_path = resource('dsl/write_pid_node.yaml')
+        dep = self.deploy(dsl_path, wait=False, client=self.client)
+        do_retries(verify_deployment_env_created, 30, deployment_id=dep.id)
+        execution = self.client.executions.start(deployment_id=dep.id,
+                                                 workflow_id='install')
+        pid = do_retries(docl.read_file, file_path='/tmp/pid.txt')
+        path = '/proc/{}/status'.format(pid)
+        execution = self.client.executions.cancel(execution.id,
+                                                  force=True, kill=True)
+        self.assertEquals(Execution.KILL_CANCELLING, execution.status)
+        # if the process is still running, docl.read_file will raise an error
```

Comment:
if the process is still running docl.read_file will raise an error. We use do_retries to give the `kill cancel` operation time to kill the process.